packages/gatsby-plugin-gatsby-cloud/src/__tests__/gatsby-browser.js

```diff
@@ -39,6 +46,40 @@ describe(`Preview status indicator`, () => {
     }
   }
 
+  const assertTrackEventGetsCalled = async ({
+    text,
+    route,
+    action,
+    testId,
+  }) => {
+    process.env.GATSBY_PREVIEW_API_URL = createUrl(route)
+    process.env.GATSBY_TELEMETRY_API = `http://test.com/events`
+    let component
+
+    jest.useFakeTimers()
+
+    await act(async () => {
+      render(<Indicator />)
+    })
+
+    await waitFor(() => {
+      if (testId) {
+        component = screen.getByTestId(testId)
+      } else {
+        component = screen.getByText(text, { exact: false })
+      }
+    })
+
+    if (action) {
+      userEvent[action](component)
+      // Initial poll fetch, initial load trackEvent, and trackEvent after action
+      expect(window.fetch.mock.calls.length).toBeGreaterThan(2)
```

Comment:
Is there a way to be explicit? greaterThan 2 doesn't view doesn't give me confidence or maybe test if it got called with the appropriate event?
```suggestion
      expect(window.fetch.mock.calls.length).toBeGreaterThan(2)
```

Reply:
Fixed it to be more precise by wrapping it in `waitFor`